cmd/kubeadm/app/componentconfigs/defaults.go

```diff
@@ -88,9 +88,11 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 	}
 
 	// Enforce security-related kubelet options
-
-	// Require all clients to the kubelet API to have client certs signed by the cluster CA
-	externalkubeletcfg.Authentication.X509.ClientCAFile = filepath.Join(internalcfg.CertificatesDir, constants.CACertName)
+	if externalkubeletcfg.Authentication.X509.ClientCAFile == "" {
+		// Require all clients to the kubelet API to have client certs signed by the cluster CA
+		// Set it to default path
+		externalkubeletcfg.Authentication.X509.ClientCAFile = filepath.Join(internalcfg.CertificatesDir, constants.CACertName)
```

Comment:
the original hardcoded default value is by design.
> // Enforce security-related kubelet options

i need to understand more about your use case.
do you want to have a client cert for the kubelet signed by a different CA or is this more of a path problem where you have the cluster CA in a different path?

@kubernetes/sig-cluster-lifecycle-pr-reviews 
/hold

Reply:
i closed this PR and reopened a new #78079. Posted my comment in the new PR